### ISSUE TITLE
Cryptol stalls when type-checking

### ISSUE THREAD

**Issue by benjaminjellison** [GaloisInc/cryptol#433]
This is similar to #432 , in that it is a weird example and it works fine ```054a3e2``` but not in ```13fef57```:

When I try to load this code into Cryptol:
```
stall = a
 where
  a = as@0
  as = [ zero : [8]] # [ i | i <- as | j <- bs | k <- cs ]
  bs = [ zero : [8] | i <- as | k <- cs]
  cs = zero
```
it stalls:
```
Loading module Cryptol
Loading module Main
cryptol: fd:16: commitBuffer: resource vanished (Broken pipe)
```
The last line only appears after waiting a while.

Again, it works fine in the older version.  Also, I wasn't able to simplify it any further.  If I take out ```i < - as```, ```j <- bs```, or ```k <- cs``` in any of the lines it loads fine.  If I change the last line to ```cs = zero : [inf][8]``` it also loads fine.  That makes me think there's something with the typechecker, although I don't know for sure.

**Comment by yav** [comment]
Hm, seems to load instantly for me.  I tried it with:
Cryptol: 771c2deaa133e01233052d236df5e48ec3a7a3e1
Z3: Z3 version 4.5.0 - 64 bit

Could you post the version of Z3 that you are using?

**Comment by benjaminjellison** [comment]
My Z3 is older: 4.3.2 - 64 bit.  I'll try upgrading my z3.

**Comment by robdockins** [comment]
@benjaminjellison, did upgrading Z3 fix your issue?

**Comment by benjaminjellison** [comment]
The problem is fixed in Cryptol 2.5, but not in ```13fef57```, even with the latest Z3.  I think you can close this issue.
